pkg/storage/snapshot/restore.go

```diff
@@ -564,18 +535,28 @@ func (t *vmRestoreTarget) reconcileSpec() (bool, error) {
 	newVM.Spec.Template.Spec.Volumes = newVolumes
 	setLastRestoreAnnotation(t.vmRestore, newVM)
 
-	if err = t.restoreInstancetypeControllerRevisions(newVM); err != nil {
+	t.updateRestoredVM(newVM)
```

Comment:
per earlier comment I think it would be better to return the new vm and not assign to a member. I just feel that mutable shared state should be avoided whenever possible

Reply:
OK